whisper_medusa/models/model.py

```diff
@@ -1458,3 +1670,65 @@ def generate(
 
         else:
             raise NotImplementedError("Longform generation is not supported yet")
+
+    def freeze_model_parts(self, parts_to_freeze):
+        self.whisper_model.freeze_model_parts(parts_to_freeze)
+
+    def set_output_whisper_original(self):
+        self.whisper_model.config.output_hidden_states = True
+        self.config.output_hidden_states = True
+        self.whisper_layer = WhisperDecoderLayer(self.whisper_model.config)
+        self.whisper_layer.load_state_dict(self.whisper_model.model.decoder.layers[-1].state_dict())  # load the last layer of the whisper model
+        for param in self.whisper_layer.parameters():
+            param.requires_grad = False
+
+
+    def output_whisper_original(self, whisper_model_outputs, use_cache, past_key_values, head_mask, attention_mask, cross_attn_head_mask):
+        with torch.no_grad():
```

Comment:
decorate the function it is more elegant.

Reply:
done